app/src/main/java/com/T05/krowdtrialz/ui/contributors/ContributorList.java

```diff
@@ -53,30 +54,29 @@ public View getView(int position, @Nullable View convertView, @NonNull ViewGroup
         User contributor = contributors.get(position);
 
         TextView contributorName = view.findViewById(R.id.contributor_name_textView);
-        CheckBox ignoreCheckBox = view.findViewById(R.id.ignore_contributor_checkbox);
+        Button ignoreButton = view.findViewById(R.id.ignore_contributor_button);
 
         contributorName.setText(contributor.getUserName());
 
-        //set checkbox if user is ignored
         if(experiment.isIgnored(contributor)){
-            ignoreCheckBox.setChecked(true);
+            ignoreButton.setText("Un-Ignore");
         } else {
-            ignoreCheckBox.setChecked(false);
+            ignoreButton.setText("Ignore");
         }
 
-        ignoreCheckBox.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+        ignoreButton.setOnClickListener(new View.OnClickListener() {
             @Override
-            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
-                if(isChecked){
-                    if(!experiment.isIgnored(contributor)){
-                        experiment.ignoreUser(contributor);
-                        db.updateExperiment(experiment);
-                    }
-                } else {
-                    if(experiment.isIgnored(contributor)){
-                        experiment.removeIgnoredUser(contributor);
-                        db.updateExperiment(experiment);
-                    }
+            public void onClick(View v) {
+                if(experiment.isIgnored(contributor)){
+                    experiment.removeIgnoredUser(contributor);
+                    db.updateExperiment(experiment);
+                    ignoreButton.setText("Ignore");
+                    Log.d(TAG, "Un-Ignored User");
+                } else{
+                    experiment.ignoreUser(contributor);
+                    db.updateExperiment(experiment);
+                    ignoreButton.setText("Un-Ignore");
+                    Log.d(TAG, "Ignored User");
```

Comment:
Can you also toggle the colour here - both when the button text is first set and in the onclick. I think orange (0xFFFF9800) would be good for whenever the box says "Ignore" and Blue (0xFF3F51B5) whenever the button says "Un-ignore".